go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/itzg/mc-server-runner
 
-go 1.17
+go 1.23
```

Comment:
Should also bump it in workflows to be consistent

https://github.com/itzg/mc-server-runner/tree/master/.github/workflows